roles/opcap_tool/tasks/build.yml

```diff
@@ -9,20 +9,15 @@
   delay: 10
 
 - name: "Use Golang container to build opcap"
-  # Launching a container to build the GO bin Dir where the repo has been
-  # cloned is bound as a volume (with option Z for SELinux compatibility)
-  containers.podman.podman_container:
-    name: go-builder
-    image: "golang:1.19"
-    debug: true
-    pids_limit: -1
-    log_level: "info"
-    command: make -C /opcap build OPCAP_VERSION={{ opcap_version }}
-    detach: false
-    tty: true
-    rm: true
-    volume:
-      - "{{ opcap_dir.path }}/opcap/:/opcap:Z"
+  # Launching a container to build the opcap binary.
+  # The directory where the repository has been cloned is bound as a volume.
+  # The 'Z' option is for SELinux compatibility.
+  command:
+    cmd: >
+      podman run --name go-builder --rm -it
+      --log-level info --pids-limit=0
+      -v {{ opcap_dir.path }}/opcap/:/opcap:Z
```

Comment:
For the selinux part, since it is a temp file, the right context could be set before running the container and avoid switching to podman

We have some other roles that do something similar that should be useful, a couple of examples:

https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/ceca24e3871bbc26b9ed85d6ee59e9d28adb32ce/roles/ocp_on_libvirt/tasks/setup.yml#L75-L83

https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/ceca24e3871bbc26b9ed85d6ee59e9d28adb32ce/roles/mirror_ocp_release/tasks/download.yml#L31-L36

Reply:
Thanks for the advice, but I'm not fixing selinux part here, I'm fixing the go build timeout. We could keep your advice for later on.